### ISSUE TITLE
[docs] Add callouts and troubleshooting sections

### ISSUE THREAD

**Issue by atomiks** [mui/base-ui#1218]
The old documentation had many of these that were useful that are missing on the new docs. We should try to port these over to the new docs (matching the new style of course).

For instance, developers are probably going to be tempted to use `Tooltip` when trying to create things like the infotips on the docsm when they should actually use `Popover` + `openOnHover`. 

No matter how much we try to optimize the API, there are going to be potential areas that may be confusing or need workarounds to document. It's a good idea to list as many of these as we can in the docs upfront to ensure devs are aware.

**Comment by mapache-salvaje** [comment]
For the specific example you mentioned: how do you feel about something like a "Usage" section for each component? We started to apply this pattern to some of the Material UI docs last year, to help users get to the correct component for their use case: https://mui.com/material-ui/react-snackbar/#usage

**Comment by atomiks** [comment]
>  to help users get to the correct component for their use case

Yeah this is what we need. I think it shouldn't be a "plain" paragraph though as in the Material UI docs - it should draw their attention more clearly by being colored or such

**Comment by battaglr** [comment]
This could also fix with #1687.

Also, this might be already solved, as some components have a "Usage guidelines" section, like the Tooltip component for example: https://base-ui.com/react/components/tooltip#usage-guidelines.
